x-pack/plugins/observability/public/data_handler.ts

```diff
@@ -28,9 +28,26 @@ export function getDataHandler<T extends ObservabilityApp>(appName: T) {
   }
 }
 
-export async function fetchHasData() {
+function getPromiseResult(promiseResult: PromiseFulfilledResult<any> | PromiseRejectedResult) {
+  if (promiseResult.status === 'fulfilled') {
+    return promiseResult.value;
+  }
+  // eslint-disable-next-line no-console
+  console.error('Error while fetching has data', promiseResult.reason);
+  return false;
+}
+
+export async function fetchHasData(): Promise<Record<ObservabilityApp, boolean>> {
   const apps: ObservabilityApp[] = ['apm', 'uptime', 'infra_logs', 'infra_metrics'];
   const promises = apps.map((app) => getDataHandler(app)?.hasData());
-  const [apm, uptime, logs, metrics] = await Promise.allSettled(promises);
-  return { apm, uptime, infra_logs: logs, infra_metrics: metrics };
+  const [apm, uptime, logs, metrics] = await Promise.allSettled(promises).then((results) =>
+    results.map(getPromiseResult)
+  );
```

Comment:
It looks like typescript return `any` when you use `then`. Using `await` will retain the type information:

```suggestion
  const results = await Promise.allSettled(promises);
  const [apm, uptime, logs, metrics] = results.map((result) => {
    if (result.status === 'fulfilled') {
      return result.value;
    }
    // eslint-disable-next-line no-console
    console.error('Error while fetching has data', result.reason);
    return false;
  });
```

Reply:
And even though it's normally good practice to split code up into smaller functions I think it's more clear to keep it inline in this case.